### ISSUE TITLE
JsonBodyFilters.replaceJsonStringProperty not replacing nested properties

### ISSUE THREAD

**Issue by chrisdruta** [zalando/logbook#787]
## Description

I configured my Logbook filtering for my servlet filter / clients with the following shared configuration:

```java
public static final Logbook LOGBOOK = Logbook.builder()
            .queryFilter(QueryFilters.replaceQuery("password", REPLACEMENT_VALUE))
            .headerFilter(HeaderFilters.replaceHeaders(HttpClientConstants.X_JWT::equalsIgnoreCase, REPLACEMENT_VALUE))
            .headerFilter(HeaderFilters.replaceHeaders(HttpClientConstants.TOKEN_HEADER_KEY::equalsIgnoreCase, REPLACEMENT_VALUE))
            .bodyFilter(BodyFilters.replaceFormUrlEncodedProperty(ImmutableSet.of("Token", "password"), REPLACEMENT_VALUE))
            .bodyFilter(JsonBodyFilters.replaceJsonStringProperty(ImmutableSet.of("apiToken", "password", "result"), REPLACEMENT_VALUE))
            .sink(new DefaultSink(
                    new JsonHttpLogFormatter(),
                    new DefaultHttpLogWriter()))
            .build();
``` 

The docs for replaceJsonStringProperty say it should replace properties at any level within the JSON tree. 

## Expected Behavior
When presented with a nested json structure containing the property `apiToken`, it's value should be replaced.

## Actual Behavior
`apiToken` value is still present and not obfuscated.

```
[https-jsse-nio-8086-exec-1] TRACE org.zalando.logbook.Logbook - 
{"origin":"remote","type":"response","correlation":"bb410499c8a93486","duration":197,"protocol":"HTTP/1.1","status":200,"body":{"result":{"apiToken":"not-obfuscated??"},"httpStatus":"200 - OK"}}}
```

Interesting enough, I tried adding the "result" property to be obfuscated as well and that failed as well.

Last ditch effort I tried separating each body filter property replacement and that didn't work either.

## Steps to Reproduce
1. Configure a logbook filter to replace custom json property
2. Execute request/response containing nested json property to be obfuscated
3. Observe not obfuscated in logs

## Context
Im just trying to do basic obfuscation

## Your Environment

* Version used: logbook-bom 2.1.0

**Comment by whiskeysierra** [comment]
Fixed with #782 but I haven't released it yet

**Comment by chrisdruta** [comment]
Cheers, cool. Hoping you release soon

**Comment by whiskeysierra** [comment]
I just released 2.1.1 with a fix.
